transform/models/marts/geo_reference/geo_reference__building_footprints_with_blocks.sql

```diff
@@ -24,9 +28,39 @@ blocks as (
 footprints_and_blocks_joined as (
     select
         footprints.*,
-        blocks.* exclude "geometry"
+        blocks.* exclude "geometry",
+        -- TODO: investigate how much savings there are by doing this with
+        -- a `case` statement rather than computing the area for everything.
+        -- also investigate whether there are any savings to be had doing
+        -- this with a self join instead of a window function.
+        case count(*) over (partition by footprints._tmp_id) > 1
+            when false then 1.0
+            when
+                true
+                then st_area(st_intersection(footprints."geometry", blocks."geometry"))
+        end as _tmp_intersection
```

Comment:
This is clever to try to do the area calculation when the count is greater than one as opposed to calculating the area for every row. It's unclear to me if / how you'd be able to evaluate if the area calculation is happening sequentially or in parallel, but I suppose one way to ensure that is to split this into two smaller CTEs. Still, I suspect based on our live discussion the alternative using max_by() will be faster.